cmapi/cmapi_server/managers/certificate.py

```diff
@@ -0,0 +1,106 @@
+"""Module related to CMAPI self-signed certificate management logic."""
+import os
+import logging
+from datetime import datetime, timedelta, timezone
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.x509.oid import NameOID
+
+
+CERT_FILENAME = './cmapi_server/self-signed.crt'
+KEY_FILENAME = './cmapi_server/self-signed.key'
+CERT_DAYS = 365
+
+
+class CertificateManager():
+    """Class with methods to manage self-signed certificate."""
+
+    @staticmethod
+    def create_self_signed_certificate() -> None:
```

Comment:
Don't we need to sync with mutex running this method when renew a cert?

Reply:
Have no idea which mutex to sync you're talking about.

Reply:
You write into a file. You can do this from multiple threads thus this is an resource access race.